auraed/src/runtime/mod.rs

```diff
@@ -143,12 +148,12 @@ impl cell_service_server::CellService for CellService {
         &self,
         request: Request<FreeCellRequest>,
     ) -> Result<Response<FreeCellResponse>, Status> {
-        // Initialize the cell
-        let r = request.into_inner();
-        let cell_name = r.cell_name;
-        info!("CellService: free() cell_name={:?}", cell_name);
-        self.remove_cgroup(&cell_name).expect("remove cgroup");
-        Ok(Response::new(FreeCellResponse {}))
+        let request = request.into_inner();
+        let request = ValidatedFreeCellRequest::validate(request, None)
+            .map_err(|e| Status::invalid_argument(e.to_string()))?;
+
+        info!("CellService: free() cell_name={:?}", &request.cell_name);
+        request.handle(self).map(Response::new)
```

Comment:
this right here echoes what i wrote above: we're now "calling handle on request passing self" instead of "calling handle on self passing request".  it feels backwards.